test/functional/rpc_rawtransaction.py

```diff
@@ -178,6 +182,67 @@ def getrawtransaction_tests(self):
         block = self.nodes[0].getblock(self.nodes[0].getblockhash(0))
         assert_raises_rpc_error(-5, "The genesis block coinbase is not considered an ordinary transaction", self.nodes[0].getrawtransaction, block['merkleroot'])
 
+    def getrawtransaction_verbosity_tests(self):
+        tx = self.nodes[2].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+        [block1] = self.generate(self.nodes[2], 1)
+        fields = [
+            'blockhash',
+            'blocktime',
+            'confirmations',
+            'hash',
+            'hex',
+            'in_active_chain',
+            'locktime',
+            'size',
+            'time',
+            'txid',
+            'vin',
+            'vout',
+            'vsize',
+            'weight'
+        ]
+        prevout_fields = [
+            'generated',
+            'height',
+            'value',
+            'scriptPubKey'
+        ]
+        script_pub_key_fields = [
+            'address',
+            'asm',
+            'hex',
+            'type'
+        ]
+        for n in [0, 3]:
+            for v in [1,2]:
+                self.log.info(f"Test getrawtransaction_verbosity {v} {'with' if n == 0 else 'without'} -txindex, with blockhash")
+                gottx = self.nodes[n].getrawtransaction(txid=tx, verbosity=v, blockhash=block1)
+                for f in fields:
+                    if f not in gottx:
+                        raise AssertionError(f"field {f} is not in transaction")
+                    if v == 1:
+                        assert('fee' not in gottx)
+                        assert(len(gottx['vin']) > 0)
+                        assert('prevout' not in gottx['vin'][0])
+                    if v == 2:
+                        assert('fee' in gottx)
+                        assert(isinstance(gottx['fee'], Decimal))
+                        assert(len(gottx['vin']) > 0)
+                        assert('prevout' in gottx['vin'][0])
+                        prevout = gottx['vin'][0]['prevout']
+                        for prevout_field in prevout_fields:
+                            if prevout_field not in prevout:
+                                raise AssertionError(f"field {prevout_field} is not in transaction prevout {prevout}")
+                        script_pub_key = prevout['scriptPubKey']
+                        for script_field in script_pub_key_fields:
+                            if script_field not in script_pub_key:
+                                raise AssertionError(f"field {script_field} is not in transaction scriptPubKey {script_pub_key}")
```

Comment:
Now these assertions are repeated for every `f in fields`, I think you want to decrease the indentation?